### PR TITLE
Css modules typescript infra via 'typescript-plugin-css-modules' + eslint ts specification

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/2861766393/pulses/3191529916

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "1.59.1",
   "description": "Official monday.com UI resources for application development in React.js",
   "main": "./dist/main.js",
+  "types": "./dist/types.d.ts",
```
Comment (orrgottlieb) [inline package.json:6]: this was removed by design

**Review by orrgottlieb — CHANGES_REQUESTED** [review]

**Review by hadasfa — CHANGES_REQUESTED** [review]
What Orr comment + it's looks like the new es lint settings cause a little bit a mess.
for example: now it's ignore _ when defined not required parameter in function and ignore // eslint-disable-next-line no-unused-vars because now I guess we don't allow it by different settings? let's make sure all the eslint errors are fixed before merging.

File: package.json (hunk 1 above)
Comment (SergeyRoyt) [reply package.json:6]: Removed 👍

File: src/components/Switch/index.js [hunk 2]
```diff
@@ -37,7 +38,7 @@ export function useHiddenSwitch({
     isChecked: overrideDefaultChecked || checked,
     HiddenSwitch: (
       <input
-        ref={ref}
+        ref={inputRef}
```
Comment (orrgottlieb) [inline src/components/Switch/index.js:41]: this might be breaking change, please use mergedRef

File: .eslintrc.js [hunk 3]
```diff
@@ -6,12 +63,32 @@ module.exports = {
...
+        "@typescript-eslint/no-unused-vars": ["error", { argsIgnorePattern: "^_", varsIgnorePattern: "^_" }]
+      }
     }
   ],
   env: {
     browser: true,
     es2021: true
   },
+  globals: {
+    require: true,
```
Comment (orrgottlieb) [inline .eslintrc.js:89]: why do we need require ?

File: src/components/Switch/index.js (hunk 2 above)
Comment (orrgottlieb) [reply src/components/Switch/index.js:41]: or was it a bug?

File: .eslintrc.js [hunk 4]
```diff
@@ -6,12 +63,32 @@ module.exports = {
...
+    module: true
```
Comment (orrgottlieb) [inline .eslintrc.js:90]: why do we need module as well?

File: src/components/Switch/index.js (hunk 2 above)
Comment (SergeyRoyt) [reply src/components/Switch/index.js:41]: It was a bug. `ref` was used, though it was not defined

File: .eslintrc.js (hunk 4 above)
Comment (SergeyRoyt) [reply .eslintrc.js:90]: image
Without it I get this kind of errors

Comment (orrgottlieb) [reply .eslintrc.js:90]: we shouldn't use them in our code, so can you please ignore them in these specific files - (these files are for scripts)

Comment (SergeyRoyt) [reply .eslintrc.js]: Done 👍

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**